### PR TITLE
Add a new "document:exists" API route

### PR DESCRIPTION
**Pull request by scottinet**
# Description

Add, document and test a new `document:exists` API route

### CODE REVIEW

File: .eslintrc.json [hunk 1]
```diff
@@ -45,6 +45,7 @@
     "no-useless-call": 2,
     "no-with": 2,
     "quotes": [2, "single"],
+    "require-atomic-updates": 0,
```
Comment (scottinet) [inline .eslintrc.json]: Until https://github.com/eslint/eslint/issues/11899 is resolved

**Comment by scottinet** [comment]
@Aschen > Done. I also activated the linter on the new features-sdk directory

File: features-sdk/fixtures/fixtures.js [hunk 2]
```diff
@@ -1,3 +1,3 @@
-https://docs.kuzzle.io/core/2/api/controllers/admin/load-fixtures/
+// https://docs.kuzzle.io/core/2/api/controllers/admin/load-fixtures/
```
Comment (alexandrebouthinon) [inline features-sdk/fixtures/fixtures.js]:
```suggestion
// See https://docs.kuzzle.io/core/2/api/controllers/admin/load-fixtures/
```

File: doc/2/api/controllers/document/exists/index.md [hunk 3]
```diff
@@ -0,0 +1,59 @@
+---
+code: true
+type: page
+title: exists
+---
+
+# exists
+
+Checks if a document exists.
```
Comment (alexandrebouthinon) [inline doc/2/api/controllers/document/exists/index.md]: Maybe there is a missing `Since 2.0.0` badge here

Comment (Aschen) [reply doc/2/api/controllers/document/exists/index.md]: We didn't add `Since 2.0.0` badge for other methods, shall we ?

Comment (scottinet) [reply doc/2/api/controllers/document/exists/index.md]: @alexandrebouthinon is right, we should add "since" badges as often as possible.

Comment (scottinet) [reply doc/2/api/controllers/document/exists/index.md]: Fixed

**Comment by scottinet** [comment]
Failing functional test is fixed by https://github.com/kuzzleio/kuzzle/pull/1515

**Pull request merged**